libsolidity/interface/Natspec.cpp

```diff
@@ -82,7 +82,11 @@ Json::Value Natspec::userDocumentation(ContractDefinition const& _contractDef)
 	{
 		string value = extractDoc(event->annotation().docTags, "notice");
 		if (!value.empty())
-			doc["events"][event->functionType(true)->externalSignature()]["notice"] = value;
+		{
+			Json::Value eventDoc{Json::objectValue};
+			eventDoc["notice"] = value;
+			doc["events"][event->functionType(true)->externalSignature()].append(std::move(eventDoc));
+		}
```

Comment:
Not entirely sure what you mean by this though - is that not how it works now (i.e. in this PR)?  It's been a while, but from what I remember, only one event's NatSpec is exported, and it's always the 'lowest' level contract in the inheritance hierarchy (i.e. the most derived one), and from what you've mentioned in chat - exporting file level events is disabled, meaning that the only thing wrong here is the fact that we're exporting the event within a list, instead of as we did before.

Reply:
Yes, I primarily mean that it's producing a list rather than a single value. This is breaking, even if the list has only one element.

> only one event's NatSpec is exported, and it's always the 'lowest' level contract in the inheritance hierarchy (i.e. the most derived one)

ok, that sounds correct, but we should still have a test case that will prove it as I mentioned in https://github.com/ethereum/solidity/pull/10996/commits/1263730c947a1da68f74f3825d994c043f11995d#r1175226720

Reply:
By the way, we should probably have some asserts checking our assumptions here if possible (or comments if not). That it's not a file-level event, an event from another contract, etc. This way of processing it will stop being correct once these are broken and we're going to break them at some point.